src/module-path.ts

```diff
@@ -1,88 +1,28 @@
 import { IHost, DefaultHost } from './host';
+import * as fs from 'fs';
+import * as browserResolve from 'browser-resolve';
 import * as nodeCoreLibs from 'node-libs-browser';
 
 /**
- * node resolve algorithms with some specialities:
- * 
- * * package.json entries are considered in this order: 'jsnext:main', 'browser', 'main'
- * * node core libraries are resolved against node-libs-browser (from webpack)
+ * Return a resolved module path
+ *
+ * @param filename Path to file from where importPath is resolved
+ * @param importIdentifier Identifier to resolve from filename
+ * @param [host]
  */
-export function getModulePath(currentModule: string, importPath: string, host: IHost = new DefaultHost()): string {
-  function notFound(): void {
-    throw new Error(`Module ${importPath} not found`);
-  }
-  function resolveAsFile(file: string): string {
-    if (host.fileExists(file) && host.isFile(file)) {
-      return file;
-    }
-    const filePath = file + '.js';
-    if (host.fileExists(filePath) && host.isFile(filePath)) {
-      return filePath;
-    }
-  }
-  function resolveAsDirectory(dir: string): string {
-    const pkgFile = host.joinPath(dir, 'package.json');
-    if (host.fileExists(pkgFile) && host.isFile(pkgFile)) {
-      const pkg = JSON.parse(host.readFile(pkgFile).toString());
-      const main = pkg['jsnext:main'] || (typeof pkg.browser === 'string' ? pkg.browser : undefined) || pkg.main;
-      if (main) {
-        const result = resolveAsFile(host.joinPath(dir, main));
-        if (result) {
-          return result;
-        }
+export function getModulePath(filename: string, importIdentifier: string, host: IHost = new DefaultHost()): string {
+  return browserResolve.sync(importIdentifier, {
+    filename: filename,
+    modules: nodeCoreLibs,
+    packageFilter: pkg => {
+      // If the browser field is not defined, use jsnext:main instead
+      if ('browser' in pkg === false && 'jsnext:main' in pkg) {
+        pkg.browser = pkg['jsnext:main'];
```

Comment:
Current strategy would be:
1. if there is a browser field use that
2. if there is a jsnext:main field use that
3. otherwise stay with default node resolve